temporalio/worker/_workflow_instance.py

```diff
@@ -1654,6 +1807,26 @@ async def handle_query(self, input: HandleQueryInput) -> Any:
         else:
             return handler(*input.args)
 
+    def handle_update_validator(self, input: HandleUpdateInput) -> None:
+        # Do not "or None" the validator, since we only want to use the validator for
+        # the specific named update - we shouldn't fall back to the dynamic validator
+        # for some defined, named update which doesn't have a defined validator.
+        handler = self._instance.workflow_get_update_validator(input.update)
+        # Validator may not be defined
```

Comment:
In theory this should not longer happen anymore since you check before interceptor, but fine to check again in here (just in case they did something wonky like unset it in an outer interceptor)